### PR TITLE
ppl: fix crash in findSlots

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/2642

### CODE REVIEW

**Comment by maliberty** [comment]
looks ok but I'll wait for the secure CI

**Comment by eder-matheus** [comment]
> looks ok but I'll wait for the secure CI

Private designs are fine; only two public designs have failures:
`bp_multi_top nangate45: [ERROR] finish__timing__drv__max_cap_limit fail test: -0.4782 >= -0.3`
`riscv32i sky130hd: [ERROR] finish__timing__drv__max_cap_limit fail test: -0.4279 >= -0.31`

I already have a branch with the updated metrics. I can create a PR for it after this one is merged.

**Review by maliberty — APPROVED** [review]
Those metrics updates look ok

**Pull request merged**
